### ISSUE TITLE
Ember Engine Support

### ISSUE THREAD

**Issue by mwalper** [Subtletree/ember-paper-link#10]
Hi,
for some unknown reason this addon throws an error when used within an ember-engine.
`
{{#paper-link-item "blog"}}...{{/paper-link-item}}`

Error:
`index.js:163 Uncaught (in promise) Error: Assertion Failed: You attempted to generate a link for the "blog" route, but did not pass the models required for generating its dynamic segments. There is no route named blog`

However, using the normal linkTo works fine.

Adding the mount point (namespace) name:
```
// router.js
this.mount('engine-name', { path: '/myengine' });
```
`
{{#paper-link-item "engine-name.blog"}}...{{/paper-link-item}}`

the error vanished.
This is fine when using the engine as an in-repo engine with only one mount point, but not with multiple mounts or as an Addon.

**Comment by Subtletree** [comment]
Thanks for reporting!

Will take a look tonight.

**Comment by Subtletree** [comment]
Looks like this is our problem: https://github.com/ember-engines/ember-engines/issues/304.

An easy solution would be to supply a separate `paper-link-engine` component that extends `ember-engines/components/link-to`.

Could also look at conditionally adding the mount namespace if we're in a routable engine.

Another would be something like https://github.com/Subtletree/ember-paper-link/issues/5#issuecomment-319502511 where we ditch `link-to` in favour of determining active state with the router service. Would also need to handle mount namespace.

Will think about it 😅

**Comment by mwalper** [comment]
Thanks for looking into it.
Any updates on this?

**Comment by Subtletree** [comment]
Added `paper-link-engine` and `paper-link-item-engine` components on master.

Can you let me know if they fix the issue for you?

I see this as a temporary work around until [the router service can handle engine routes](https://github.com/ember-engines/ember-engines/issues/587). I think then we can skip `link-to` and use the route service instead.

**Comment by Subtletree** [comment]
@mwalper let me know how the above goes for you and I'll cut a new release 🙂

**Comment by Subtletree** [comment]
Now available in v1.0.0. Currently undocumented..
